PeerioTechnologies/peerio-icebear
# feat: new contact color calculations

#### Relevant info and issue/PR links 
CH https://app.clubhouse.io/peerio/story/8090/client-sdk-update-default-avatar-colors
PeerUI https://github.com/PeerioTechnologies/ui-library/pull/4
desktop https://github.com/PeerioTechnologies/peerio-desktop/pull/351
 
#### Testing instructions  
Design has requested new avatar for when user has:
* instead of being totally random, we select from 36 preset colours that are consistent with design's guidelines
* `firstLetter` shows either black or white text depending on the color

This is a breaking change because `color` no longer returns the `#000000` string, instead it returns an object that contains `{value: '#000000', isLight: true}`. So we'll need to update desktop, PeerUI, and mobile all at the same time.

The alternative is to create a new `colorObject`, and then change computed `color` to check `colorObject`, and create new computed `colorIsLight` which also checks `colorObject`. This would be a non-breaking change, but imo would be uglier on icebear.

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
